odoo/models.py

```diff
@@ -3019,6 +3007,9 @@ def _condition_to_sql(self, alias: str, fname: str, operator: str, value, query:
                     return SQL("(%s IS NULL)", sql_field)
 
             elif isinstance(value, SQL):
+                if value.code[:1] != '(':
+                    # add parentheses if we didn't get any
+                    value = SQL("(%s)", value)
```

Comment:
In this context we need parentheses, but the person who calls the function should not be aware of this.
We could always add parentheses, however I added the `if` so that we have less of them in the output.

Also, not doing this dynamically would result in more changes because `inselect` was adding parentheses always and `in` is not adding them for SQL.

Reply:
> Also, not doing this dynamically would result in more changes because inselect was adding parentheses always and in is not adding them for SQL.

Indeed, but I rather prefer make that change now and don't have a magic behavior (adding parentheses when the first character is not a parenthesis)